doc/whats_new/v1.0.rst

```diff
@@ -164,6 +164,15 @@ Changelog
   Use ``var_`` instead.
   :pr:`18842` by :user:`Hong Shao Yang <hongshaoyang>`.
 
+:mod:`sklearn.neighbors`
+..........................
+
+- |Enhancement| Improve :class:`neighbors.KDTree` worst-case time complexity
+  from :math:`\mathcal{O}(n^2)` to :math:`\mathcal{O}(n \log n)` on sorted and
```

Comment:
The worst-case complexity is O(n), not O(n log n). 

If it is O(n log n) there is a bug in C++ STL implementation. The C++ standard requires std::nth_element to be O(n) worst-case, not just O(n) on average.